components/Vote/ElectionCandidacy.js

```diff
@@ -0,0 +1,540 @@
+      (
+        !value &&
+        t('Account/AddressForm/city/error/empty')
+      )
+    )
+  },
+  {
+    label: t('Account/AddressForm/country/label'),
+    name: 'country',
+    validator: (value) => (
+      (
+        !value &&
+        t('Account/AddressForm/country/error/empty')
+      )
+    )
+  }
+]
+
+const fields = (t) => ([
+  {
+    label: 'Statement (Ihre Motivation)',
+    name: 'statement',
+    autoSize: true,
+    validator: value =>
+      (!value && 'Statement fehlt') ||
+      (value.trim().length >= 140 && t('profile/statement/tooLong'))
+  },
+  {
+    label: t('Account/Update/birthday/label'),
+    name: 'birthday',
+    mask: '11.11.1111',
+    maskChar: '_',
+    validator: (value) => {
+      const parsedDate = birthdayParse(value)
+      return (
+        (
+          parsedDate === null ||
+          parsedDate > (new Date()) ||
+          parsedDate < (new Date(1798, 3, 12))
+        ) &&
+        t('Account/Update/birthday/error/invalid')
+      )
+    }
+  },
+  {
+    label: 'Funktion (Tätigkeit, Beruf, Amt)',
+    name: 'credential',
+    validator: (value) => {
+      return (
+        ((!value || value === '') & 'Funktion fehlt') ||
+        (value.trim().length >= 40 && t('profile/credentials/errors/tooLong'))
+      )
+    }
+  },
+  {
+    label: t('profile/disclosures/label'),
+    name: 'disclosures',
+    autoSize: true
+  }
+])
+
+const styles = {
+  previewWrapper: css({
+    margin: '20px 0'
+  }),
+  vSpace: css({
+    marginTop: 20
+  }),
+  section: css({
+    marginTop: 40
+  }),
+  error: css({
+    color: colors.error
+  }),
+  saveButton: css({
+    textAlign: 'center',
+    width: 300,
+    height: 60,
+    [mediaQueries.onlyS]: {
+      width: '100%'
+    }
+  })
+}
+
+class ElectionCandidacy extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      isEditing: props.url.query.hasOwnProperty('edit') || false,
+      showErrors: true,
+      errors: {},
+      dirty: {},
+      ...this.deriveStateFromProps(props)
+    }
+
+    this.startEditing = () => {
+      this.setState({isEditing: true})
+    }
+
+    this.save = () => {
+      const {updateCandidacy, me} = this.props
+      const { values } = this.state
+
+      this.setState({updating: true})
+
+      return updateCandidacy({
+        slug: ELECTION_SLUG,
+        statement: values.statement,
+        credential: values.credential,
+        disclosures: values.disclosures,
+        birthday: values.birthday && values.birthday.length
+          ? values.birthday.trim()
+          : null,
+        portrait: values.portraitPreview ? values.portrait : undefined,
+        address: {
+          name: me.name,
+          line1: values.line1,
+          line2: values.line2,
+          postalCode: values.postalCode,
+          city: values.city,
+          country: values.country
+        }
+      }).then(() => {
+        return new Promise(resolve => setTimeout(resolve, 400))
```

Comment:
makes the UI less jumpy

Reply:
Now I get why: because during those 400ms the refetch query happens. The clean way would be to not use refetch query and instead query the modified data in the mutation.

Reply:
Not really. The reason was that the server roundtrip is so quick that the UI changes too quickly. I put in a 400ms delay to slow things down and show the spinner.